### PR TITLE
Refactored all single record actions

### LINKED ISSUE #8973: Refactor RecordShowPageWorkflowHeader to use contextual actions
## Scope & Context

Refactor RecordShowPageWorkflowHeader to use contextual actions.
The actions on `RecordShowPageWorkflowHeader` have been hardcoded and the header should be the same as on any other Showpage.

### PR DESCRIPTION
**Pull request by bosiraphael**
## Context

Refactored all single record actions so they can be defined by a config file.
This refactoring is made with the idea that later the actions will be stored in the database, so we needed a way to serialize them.
For each object we can define a config file, if an object has no config file, it falls back to the default config.
I introduced action hooks, which return:
- `shouldBeRegistered`: `boolean` Whether the action should be registered.
- `onClick`: `() => void` The code that will be executed when we click on an action
- `ConfirmationModal`?: `React.ReactNode` (optional) The confirmation modal which will be displayed on click

This PR also closes #8973 

## Next steps

- Refactor multiple records actions
- Refactor no selection actions
- Add tests

### CODE REVIEW

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/constants/DefaultSingleRecordActionsConfigV2.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { useRemoveFromFavoritesSingleRecordAction } from '@/action-menu/actions/record-actions/single-record/hooks/useRemoveFromFavoritesSingleRecordAction';
+import { SingleRecordActionHook } from '@/action-menu/actions/types/singleRecordActionHook';
+import {
+  ActionMenuEntry,
+  ActionMenuEntryScope,
+  ActionMenuEntryType,
+} from '@/action-menu/types/ActionMenuEntry';
+import { IconHeart, IconHeartOff, IconTrash } from 'twenty-ui';
+
+export const DEFAULT_SINGLE_RECORD_ACTIONS_CONFIG_V2: Record<
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/constants/DefaultSingleRecordActionsConfigV2.ts:12]: sorry what is the diff with V1?

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useAddToFavoritesSingleRecordAction.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { SingleRecordActionHookWithObjectMetadataItem } from '@/action-menu/actions/types/singleRecordActionHook';
+import { useCreateFavorite } from '@/favorites/hooks/useCreateFavorite';
+import { useFavorites } from '@/favorites/hooks/useFavorites';
+import { recordStoreFamilyState } from '@/object-record/record-store/states/recordStoreFamilyState';
+import { useRecoilValue } from 'recoil';
+import { isDefined } from 'twenty-ui';
+
+export const useAddToFavoritesSingleRecordAction: SingleRecordActionHookWithObjectMetadataItem =
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useAddToFavoritesSingleRecordAction.ts:8]: please test this hook. You removed tests that will drop coverage

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useRemoveFromFavoritesSingleRecordAction.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { SingleRecordActionHookWithObjectMetadataItem } from '@/action-menu/actions/types/singleRecordActionHook';
+import { useDeleteFavorite } from '@/favorites/hooks/useDeleteFavorite';
+import { useFavorites } from '@/favorites/hooks/useFavorites';
+import { isDefined } from 'twenty-ui';
+
+export const useRemoveFromFavoritesSingleRecordAction: SingleRecordActionHookWithObjectMetadataItem =
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useRemoveFromFavoritesSingleRecordAction.ts:6]: same !

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/workflow-version-actions/hooks/useSeeWorkflowVersionsHistoryWorkflowVersionSingleRecordAction.ts [hunk 4]
```diff
@@ -1,27 +1,24 @@
...
 
-export const useSeeWorkflowVersionsHistoryWorkflowVersionSingleRecordAction = ({
-  workflowVersionId,
-}: {
-  workflowVersionId: string;
-}) => {
-  const workflowVersion = useRecoilValue(
-    recordStoreFamilyState(workflowVersionId),
-  );
+export const useSeeWorkflowVersionsHistoryWorkflowVersionSingleRecordAction: SingleRecordActionHookWithoutObjectMetadataItem =
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/workflow-version-actions/hooks/useSeeWorkflowVersionsHistoryWorkflowVersionSingleRecordAction.ts:7]: please rename, I think useSeeWorkflowVersionsHistorySingleRecordAction is already long :D

**Review by thomtrp — COMMENTED** [review]
Please add tests for your hooks. Otherwise someone else will be stuck with coverage soon :)

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/components/SingleRecordActionMenuEntrySetterEffect.tsx [hunk 5]
```diff
@@ -1,24 +1,81 @@
...
+
+  const actionConfig =
+    objectMetadataItem.nameSingular === CoreObjectNameSingular.Workflow
+      ? WORKFLOW_SINGLE_RECORD_ACTIONS_CONFIG
+      : objectMetadataItem.nameSingular ===
+          CoreObjectNameSingular.WorkflowVersion
+        ? WORKFLOW_VERSIONS_SINGLE_RECORD_ACTIONS_CONFIG
+        : isPageHeaderV2Enabled
+          ? DEFAULT_SINGLE_RECORD_ACTIONS_CONFIG_V2
+          : DEFAULT_SINGLE_RECORD_ACTIONS_CONFIG_V1;
```
[the remark this answers is not in the document]
Comment (bosiraphael) [reply packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/components/SingleRecordActionMenuEntrySetterEffect.tsx:31]: There are no actions implemented for the moment for the workflow runs ! But when there will be yes ! :)

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/constants/DefaultSingleRecordActionsConfigV2.ts (hunk 1 above)
Comment (bosiraphael) [reply packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/constants/DefaultSingleRecordActionsConfigV2.ts:12]: Pinned actions are different

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/components/SingleRecordActionMenuEntrySetterEffect.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/components/SingleRecordActionMenuEntrySetterEffect.tsx:31]: ternary is too complex, extract to an util that will return early to eliminate cases progressively

File: packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useAddToFavoritesSingleRecordAction.ts (hunk 2 above)
Comment (bosiraphael) [reply packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useAddToFavoritesSingleRecordAction.ts:8]: I have another ticket in this sprint to add/update the tests

**Review by charlesBochet — COMMENTED** [review]
I haven't tested it functionnally but the code looks great, love the approach
Please put back some tests!

**Review by thomtrp — APPROVED** [review]

**Pull request merged**